src/connection/connection-manager.ts

```diff
@@ -0,0 +1,58 @@
+import type { Dwn } from "@tbd54566975/dwn-sdk-js";
+
+import type { IncomingMessage } from "http";
+import type { WebSocket } from 'ws';
+
+import { SocketConnection } from "./socket-connection.js";
+
+/**
+ * Interface for managing `WebSocket` connections as they arrive.
+ */
+export interface ConnectionManager {
+  /** connect handler used for the `WebSockets` `'connection'` event. */
+  connect(socket: WebSocket, request?: IncomingMessage): Promise<void>;
+  /** cleans up handlers associated with the `WebSocket` connection */
+  close(socket:WebSocket): Promise<void>;
+  /** closes all of the connections */
+  closeAll(): Promise<void>
```

Comment:
Depending on the other comment on whether we have access to individual sockets when close even is fired or not, there may be no point having both `close()` and `closeAll()` at the interface level.

Reply:
Yup! Removed close(socket);